firmware/library/L1_Drivers/dac.hpp

```diff
@@ -1,19 +1,76 @@
+// Dac is a driver for the Digital-To-Analog converter for the LPC40xx chip
 #pragma once
-
-#include <cstdint>
-#include <cstdio>
+#include "L0_LowLevel/LPC40xx.h"
+#include "L1_Drivers/pin_configure.hpp"
 
 class DacInterface
 {
  public:
-    virtual bool Initialize();
-    virtual bool WriteDac(uint16_t dac_output);
-    virtual bool SetVoltage(float voltage);
-    // Sine Demonstration Function
-    virtual void SineDac(uint32_t frequency_hz);
-    // Sawtooth Demonstration Function
-    virtual void SawtoothDac(uint32_t frequency_hz);
-    // Triangle Demonstration Function
-    virtual void TriangleDac(uint32_t frequency_hz);
-    virtual void SetBias(bool bias_level);
+    enum BiasLevel: uint8_t
+    {
+        kBiasHigh  = 0,
+        kBiasLow   = 1
+    };
+    virtual void Initialize(void)                   = 0;
+    virtual bool WriteDac(uint16_t dac_output)      = 0;
+    virtual bool SetVoltage(float voltage)          = 0;
+    virtual void SetBias(BiasLevel bias_level)      = 0;
+};
+class Dac : public DacInterface
+{
+ public:
+    static LPC_DAC_TypeDef * dac_ptr;
+    static LPC_SC_TypeDef * sc_ptr;
+    static LPC_IOCON_TypeDef * iocon_ptr;
+    static constexpr float kVref = 3.3;
+    enum Bit : uint8_t
+    {
+        kClockDividerReg = 0,
+        kBiasReg = 16,
+        kDacOutReg = 6,
+        kEnableDacReg = 16
+    };
+    // PinConfigure led3(PinConfigure::CreatePinConfigure<1, 15>());
+    /* Initialize Dac by setting the clock divider and enabling 
+      The Dac Pin withiin IOCON.  Then set the initial Bias level to 0 */
+    void Initialize(void) override
+    {
+        // Set Clock Divider to 1
+        sc_ptr->PCLKSEL = (1 << kClockDividerReg);
+        // Enable DAC Pin P0.26
+        iocon_ptr->P0_26 = (1 << kEnableDacReg);
+        // Set Update Rate to 1MHz
+        dac_ptr->CR &= ~(1 << kBiasReg);
+    }
+    // The Dac ouptut is a 10 bit input and thus it is necessary to
+    // ensure dac_output is less than 1024(largest 10-bit number)
+    bool WriteDac(uint16_t dac_output) override
+    {
+        SJ2_ASSERT_FATAL(dac_output > 1023,
+                         "dac_output is above SJ2"
+                         "output capacity.");
```

Comment:
You can be more specific here. For example you can say, `Input invalid. Input MUST be between 0 and 1023`. So that the user immediately knows what they need to fix. Otherwise, the developer will have to look up what the SJ2's output capacity is.